### PR TITLE
Support a flag to find config file

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
**flyte-cli** looks to `~/.flyte/config` as the default place to find a configuration file. This is good, but we should also support a flag to allow the user to specify a different one. This is helpful for developers of Flyte, who might have several configuration files pointing to different Admins, and advanced users who might be running a local Flyte cluster as well as a live production (and staging) cluster.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
If users don't specific conf, flyte will use default config.
```
(flyte) ➜  flytekit git:(sub-flag) ✗ flyte-cli --config ~/.flyte/config get-task
Using config file at /home/kevin/.flyte/config
Usage: flyte-cli get-task [OPTIONS]
Try 'flyte-cli get-task --help' for help.

Error: Missing option '-u' / '--urn'.
(flyte) ➜  flytekit git:(sub-flag) ✗ flyte-cli --config ~/.flyte/xxxx get-task  
Config file not found at /home/kevin/.flyte/xxxx
Usage: flyte-cli get-task [OPTIONS]
Try 'flyte-cli get-task --help' for help.

Error: Missing option '-u' / '--urn'.
(flyte) ➜  flytekit git:(sub-flag) ✗ flyte-cli get-task                         
Using default config file at /home/kevin/.flyte/config
Usage: flyte-cli get-task [OPTIONS]
Try 'flyte-cli get-task --help' for help.
```
```
Usage: flyte-cli [OPTIONS] COMMAND [ARGS]...

  Command line tool for interacting with all entities on the Flyte Platform.

Options:
  -i, --insecure      Do not use SSL  [required]
  -n, --name TEXT     [Optional] The name to pass to the sub-command (if
                      applicable)  If set again in the sub-command, the sub-
                      command's parameter takes precedence.

  -d, --domain TEXT   [Optional] The domain to pass to the sub-command (if
                      applicable)  If set again in the sub-command, the sub-
                      command's parameter takes precedence.

  -p, --project TEXT  [Optional] The project to pass to the sub-command (if
                      applicable)  If set again in the sub-command, the sub-
                      command's parameter takes precedence.

  -h, --host TEXT     [Optional] The host to pass to the sub-command (if
                      applicable).  If set again in the sub-command, the sub-
                      command's parameter takes precedence.

  -c, --config TEXT   [Optional] The config to pass to the sub-command (if
                      applicable).  If set again in the sub-command, the sub-
                      command's parameter takes precedence.

  --help              Show this message and exit.
```

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1036

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/clis/flyte_cli/main.py [hunk 1]
```diff
@@ -529,6 +527,14 @@ def make_context(self, cmd_name, args, parent=None):
         return ctx
 
 
+@_click.option(
+    *_CONFIG_FLAGS,
+    required=False,
+    type=str,
+    default=None,
+    help="[Optional] The config to pass to the sub-command (if applicable).  If set again in the sub-command, "
```
Comment (katrogan) [inline flytekit/clis/flyte_cli/main.py:535]:
```suggestion
    help="[Optional] The filepath to the config file to pass to the sub-command (if applicable).  If set again in the sub-command, "
```

Comment (pingsutw) [reply flytekit/clis/flyte_cli/main.py]: Updated it, thanks.
